src/Features/Core/Shared/Utilities/LinkedFilesSymbolEquivalenceComparer.cs

```diff
@@ -16,7 +17,14 @@ internal sealed class LinkedFilesSymbolEquivalenceComparer : IEqualityComparer<I
 
         bool IEqualityComparer<ISymbol>.Equals(ISymbol x, ISymbol y)
         {
-            return x.Kind == y.Kind && x.Name == y.Name;
+            return KindsAreEquivalent(x.Kind, y.Kind) && x.Name == y.Name;
+        }
+
+        private bool KindsAreEquivalent(SymbolKind x, SymbolKind y)
+        {
+            return x == y  
+                   || (x == SymbolKind.Property && y == SymbolKind.Field)
+                   || (x == SymbolKind.Field && y == SymbolKind.Property);
```

Comment:
In cases like this, where some symbol appears with the same text, we could consider not showing the triangle but updating the tooltip to explain what symbol was found in each head.

For now, I'll go ahead and check only for name identity.